src/megatron/bridge/models/conversion/peft_bridge.py

```diff
@@ -0,0 +1,531 @@
+                adapter_task.linear_in_task.param_weight, adapter_task.linear_in_task.megatron_module
+            )
+            linear_in_tensor = next(iter(linear_in_dict.values()))
+
+            linear_out_dict = adapter_task.linear_out_task.mapping.megatron_to_hf(
+                adapter_task.linear_out_task.param_weight, adapter_task.linear_out_task.megatron_module
+            )
+            linear_out_tensor = next(iter(linear_out_dict.values()))
+
+            materialized.append(
+                AdapterWeight(
+                    global_base_prefix=adapter_task.global_base_prefix,
+                    adapter_key=adapter_task.adapter_key,
+                    alpha=adapter_task.alpha,
+                    dim=adapter_task.dim,
+                    linear_in_weight=MegatronWeightTuple(
+                        adapter_task.linear_in_task.param_name,
+                        linear_in_tensor,
+                        adapter_task.linear_in_task.vp_stage,
+                    ),
+                    linear_out_weight=MegatronWeightTuple(
+                        adapter_task.linear_out_task.param_name,
+                        linear_out_tensor,
+                        adapter_task.linear_out_task.vp_stage,
+                    ),
+                )
+            )
+
+        return materialized
+
+    def stream_adapter_weights_megatron_to_hf(
+        self,
+        megatron_model: Union[MegatronModel, List[MegatronModel]],
+        cpu: bool = True,
+        show_progress: bool = True,
+    ) -> Iterable[HFWeightTuple]:
+        """Stream only adapter weights without merging them into base tensors."""
+
+        if not isinstance(megatron_model, list):
+            megatron_model = [megatron_model]
+
+        adapter_tasks_by_base = self.build_adapter_conversion_tasks(megatron_model)
+        adapter_tasks = list(itertools.chain.from_iterable(adapter_tasks_by_base.values()))
+        if not adapter_tasks:
+            return
+
+        for adapter_task in self._with_progress_tracking(adapter_tasks, "Streaming adapter weights", show_progress):
+            adapter_weight = self.materialize_adapter_weights([adapter_task])[0]
+
+            linear_in_tensor = adapter_weight.linear_in_weight.weight
+            linear_out_tensor = adapter_weight.linear_out_weight.weight
+
+            if cpu:
+                linear_in_tensor = linear_in_tensor.cpu()
+                linear_out_tensor = linear_out_tensor.cpu()
+
+            linear_in_hf_name = adapter_task.linear_in_task.mapping.hf_param
+            linear_out_hf_name = adapter_task.linear_out_task.mapping.hf_param
+
+            assert isinstance(linear_in_hf_name, str) and isinstance(linear_out_hf_name, str), (
+                "Adapter mappings must resolve to string HF parameter names"
+            )
+
+            yield HFWeightTuple(linear_in_hf_name, linear_in_tensor)
+            yield HFWeightTuple(linear_out_hf_name, linear_out_tensor)
```

Comment:
We shouldn't make `from megatron.bridge.models.conversion.model_bridge import HFWeightTuple` only called under `TYPE_CHECKING`, as here we actually need new instances of `HFWeightTuple`. We could probably import `HFWeightTuple` inside the function if that's for preventing circular imports.